src/core/model_repository_manager.cc

```diff
@@ -1164,21 +1167,44 @@ ModelRepositoryManager::LoadModelByDependency()
     // load valid models and wait for load results
     std::vector<std::unique_ptr<ModelState>> model_states;
     for (auto& valid_model : set_pair.first) {
-      std::string repository_path;
-      const auto itr = infos_.find(valid_model->model_name_);
-      repository_path = itr->second->model_repository_path_;
-
       model_states.emplace_back(new ModelState(valid_model));
       auto model_state = model_states.back().get();
-      std::set<int64_t> versions;
-      Status status;
-      status = VersionsToLoad(
-          repository_path, valid_model->model_name_, valid_model->model_config_,
-          &versions);
+      const auto itr = infos_.find(valid_model->model_name_);
+      // FIXME clean up version logic and move agent invocation into
```

Comment:
It was supposed to be a separate enhancement. But it wouldn't be true if we want the lifecycle of the AgentModel to tie with the model, then it will be part of the BackendLifeCycle